pallets/balances/src/lib.rs

```diff
@@ -507,6 +507,11 @@ impl<T: Trait> Module<T> {
         // Commented out for now - but consider it instructive.
         // assert!(!Self::total_balance(who).is_zero());
         <FreeBalance<T>>::insert(who, balance);
+
+        if balance == 0u128.into() {
+            T::OnFreeBalanceZero::on_free_balance_zero(who);
```

Comment:
I don't think we should be adding this here. We do not want to reap accounts based on their balance. This can lead to some modules deleting important data (like staking module deleting stash data, contracts module deleting the contract data), etc.